pytest-selenium/pages/osweb.py

```
@@ -1,14 +1,112 @@
+    _user_nav_locator = (By.CSS_SELECTOR, '[class*="login-menu"]')
+    _logout_locator = (By.CSS_SELECTOR, "[href*=logout]")
+    _mobile_user_nav_locator = (By.CSS_SELECTOR, '[aria-label="Toggle Meta Navigation Menu"]')
+    _close_locator = (By.CSS_SELECTOR, '[class="close"]')
+    _mobile_user_nav_loaded_locator = (By.CSS_SELECTOR, '[aria-expanded="true"]')
+    _view_online_desktop_locator = (
+        By.XPATH,
+        "//div[@class='bigger-view']//span[text()='View online']/..",
+    )
+    _view_online_mobile_locator = (
+        By.XPATH,
+        "//div[@class='phone-view']//span[text()='View online']/..",
+    )
 
     @property
     def loaded(self):
         """Return True when the page-loaded class is added to the body tag."""
-        return (self.find_element(*self._root_locator)).is_displayed()
+        return (self.find_element(*self._root_locator)).is_displayed() and not self.find_elements(
+            *self._async_hide_locator
+        )
+
+    @property
+    def login(self):
+        return self.find_element(*self._login_locator)
+
+    @property
+    def user_nav(self):
+        return self.find_element(*self._user_nav_locator)
+
+    def mobile_user_nav_loaded(self):
+        return self.find_element(*self._mobile_user_nav_loaded_locator).is_displayed()
+
+    @property
+    def mobile_user_nav(self):
+        return self.find_element(*self._mobile_user_nav_locator)
+
+    @property
+    def user_is_logged_in(self):
+        if self.is_desktop:
+            if self.is_element_present(*self._user_nav_locator):
+                return True
+        elif self.is_mobile:
+            self.mobile_user_nav.click()
+            if self.is_element_present(*self._mobile_user_nav_locator):
+                self.mobile_user_nav.click()
+                self.wait.until(
+                    expected.invisibility_of_element_located(self._mobile_user_nav_loaded_locator)
+                )
+                return True
```

This should probably be in a `try:` block. If the `wait` fails, instead of returning `False` (not logged in) you'll get a `TimeoutException`.